modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/task/CreateClientExtensionConfigTask.java

```diff
@@ -491,6 +495,48 @@ private boolean _isWildcardValue(String value) {
 		return false;
 	}
 
+	private void _mapGlobalJSScriptElementAttributesToString(
+		ClientExtension clientExtension) {
+
+		Map<String, Object> typeSettings = clientExtension.typeSettings;
+
+		Map<String, Object> scriptElementAttributes =
+			(Map<String, Object>)typeSettings.get("scriptElementAttributes");
+
+		if (scriptElementAttributes == null) {
+			return;
+		}
+
+		Set<Map.Entry<String, Object>> entrySet =
+			scriptElementAttributes.entrySet();
+
+		Stream<Map.Entry<String, Object>> stream = entrySet.stream();
+
+		typeSettings.put(
+			"scriptElementAttributes",
+			stream.map(
+				entry -> {
+					StringBuilder stringBuilder = new StringBuilder();
+
+					stringBuilder.append(entry.getKey());
+					stringBuilder.append("=\"");
+
+					Object value = entry.getValue();
+
+					if (value == null) {
```

Comment:
To add my 2c: 

From the perspective of workspace, I would prefer to keep the values explicit. It seems odd to me that these two keys are both boolean with opposite values:

```yaml
foo:
bar: false
```

IMO this is a hidden API and is not consistent with what we see elsewhere in the YAML files. It seems like the idea is to emulate a command-line flag like `--foo`, which _does_ mean true in that context, but it does not translate well to `yaml` in my opinion. I think we should go with clarity over brevity for this case.

Reply:
I guess that's fine:

From the example you mentioned:

```
scriptElementAttributes
  defer: true // defer
  defer: false //
  defer: //

  defer: "true" // defer="true"
  defer: "false" // defer="false"
  defer: "" // defer=""
  defer: "defer" // defer="defer"
```
that's going to be translated to the corresponding values after `//` in the script element. But what's going to be shown in the UI?

**Table 1**
| Attribute 	| Value 	|
|-----------	|-------	|
| defer     	|       	|
| defer     	| false 	|
| defer     	| null  	|
| defer     	| true  	|
| defer     	| false 	|
| defer     	|       	|
| defer     	| defer 	|

That's confusing, the user doesn't know what's a string or boolean.

### Proposal

My proposal is to differentiate boolean and string attributes. So, by creating the CX from the UI, the user passes

**Table 2**
| Attribute 	| Value   	|
|-----------	|---------	|
| defer     	|         	|
| defer     	| false   	|
| defer     	| null    	|
| defer     	| "true"  	|
| defer     	| "false" 	|
| defer     	| ""      	|
| defer     	| "defer" 	|

that's stored in the CX as:

`defer defer=false defer=null defer="true" defer="false" defer="" defer="defer"`

when being added to the script element, we apply the exceptions (which is not adding falsy values)

(~~defer=false~~ ~~defer=null~~)

`<script defer defer="true" defer="false" defer="" defer="defer"/>`

For the YAML example, the behavior would be the same, it would be translated to what we have in Table 2 on the UI (.

```
scriptElementAttributes
  defer: true // defer
  defer: false //
  defer: //

  defer: "true" // defer="true"
  defer: "false" // defer="false"
  defer: "" // defer=""
  defer: "defer" // defer="defer"
```

Reply:
One change, based on discussing this with @drewbrokke , when `defer:` is passed we should throw an error since it's ambiguous whether it makes more sense to interpret null as true or false. So, throwing an error requiring that boolean values pass true or false will save the developer from wondering.

Reply:
Awesome

For the record
```
defer: // throws an error "A value for the attribute 'defer' must be specified"
defer: false // fine
defer: true // fine
```

Reply:
One more thought here - is `defer: false` useful at all? Seems kinda redundant, like the only _really_ useful form is `defer: true`.

Reply:
False let’s us ensure that the property can’t be set to true when it’s being assigned to a page.

Reply:
Ah, makes sense. Thanks @ethib137

Reply:
I updated the PR with the changes from this conversation. Be in mind that now 

```
attribute1: null
attribute2:
```

are both invalid for the same reason.

Thanks @ethib137 @drewbrokke